src/utility.js

```diff
@@ -200,3 +204,54 @@ function countFlags(value) {
 	}
 	return count;
 }
+
+/**
+ * Compute compensating position offsets if body or html element has non-static position attribute.
+ * @private
+ * @param {number} windowWidth Window width in pixels.
+ * @param {number} windowHeight Window height in pixels.
+ * @return {Offsets} The top, left, right, bottom offset in pixels
+ */
+function computePositionCompensation(windowWidth, windowHeight) {
+	// Check if the element is "positioned". A "positioned" element has a CSS
+	// position value other than static. Whether the element is positioned is
+	// relevant because absolutely positioned elements are positioned relative
+	// to the first positioned ancestor rather than relative to the doc origin.
+	var isPositioned = function(el) {
+		return el.css('position') !== 'static';
+	};
+
+	var getElementOffsets = function(el) {
+		var elWidthWithMargin,
+			elHeightWithMargin,
+			elPositionPx,
+			offsets;
+		// jquery offset and position functions return top and left
+		// offset function computes position + margin
+		offsets = el.offset();
+		elPositionPx = el.position();
+
+		// Compute the far margins based off the outerWidth values.
+		elWidthWithMargin = el.outerWidth(true);
+		elHeightWithMargin = el.outerHeight(true);
+
+		// right offset = right margin + body right position
+		offsets.right = (elWidthWithMargin - el.outerWidth() - (offsets.left - elPositionPx.left)) + (windowWidth - elWidthWithMargin - elPositionPx.left);
+		// bottom offset = bottom margin + body bottom position
+		offsets.bottom = (elHeightWithMargin - el.outerHeight() - offsets.top) + (windowHeight - elHeightWithMargin - elPositionPx.top);
```

Comment:
Are the `elPositionPx` values actually needed? They seems to be causing a bug (I'll detail in PR comments).